Model/Service/Recommendation/Category.php

```diff
@@ -77,27 +77,22 @@ public function getSortedProductIds(
         if (!$featureAccess->canUseGraphql()) {
             return $productIds;
         }
-
-        switch ($type) {
-            case NostoHelperSorting::NOSTO_PERSONALIZED_KEY:
-                $recoOperation = new CategoryBrowsingHistory($nostoAccount, $nostoCustomerId);
-                break;
-            default:
-                $recoOperation = new CategoryTopList($nostoAccount, $nostoCustomerId);
-                break;
+        if ((string)$type === NostoHelperSorting::NOSTO_PERSONALIZED_KEY) {
```

Comment:
Why do we need to cast this?

Reply:
We're always returning a string in the code, but there's no type enforcement in any of the function calls. So i've added a casting just to be sure.